### ISSUE TITLE
[pickers] 

### ISSUE THREAD

**Issue by oliviertassinari** [mui/mui-x#10048]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

It would be nice not to see the cursor on the date field.

I don't know how much effort this is, so to weigh, it's not a huge UX improvement, but I think it would feel better, based on playing with the Chrome native date picker and React Aria.

### Examples 🌈

On https://react-spectrum.adobe.com/react-spectrum/DatePicker.html#datepicker, you don't see the mouseDown caret, it selects the relevant section directly

https://github.com/mui/mui-x/assets/3165635/0dbd34d4-ad70-478b-b960-a41f7a629504

On https://mui.com/x/react-date-pickers/date-picker/, you see it, it feels a bit strange

https://github.com/mui/mui-x/assets/3165635/ad6ee935-4ceb-4cb2-ad3a-b376bd93f91d

**Comment by noraleonte** [comment]
@oliviertassinari I agree that the behavior seems a bit strange 🤔 The first section is selected on mouse down/ on focus, but then you have to wait for mouse up for the next section that you clicked on to be highlighted🤔 I am not sure about the drawback of not being able to select multiple sections, however... I'll add this to the list of items that need grooming to discuss it in more detail with the team 👍

**Comment by gitstart** [comment]
@oliviertassinari we would like to pick this up

**Comment by LukasTy** [comment]
@gitstart I've assigned this issue to you, feel free to explore a solution.
We'll see if and then it would be most suitable to apply, given that it might be a breaking change. 👌

**Comment by TheOneTheOnlyJJ** [comment]
Any update on this?

**Comment by LukasTy** [comment]
@gitstart did you have a chance to investigate this issue? 🤔 
I added a `v8.x` label as I suspect this change would have some breaking behaviors that would block us from delivering it in the current major.

**Comment by LukasTy** [comment]
Such a behavior is already present on the [accessible DOM structure](https://mui.com/x/react-date-pickers/fields/#accessible-dom-structure).

Given that it will become the default on v8, there is no clear rushing factor to implement it on the input behavior.
Especially, when the same behavior can be seen on other implementations using input (i.e. [Kendo](https://www.telerik.com/kendo-react-ui/components/dateinputs/dateinput/)).

**Comment by oliviertassinari** [comment]
> it will become the default on v8

@LukasTy Could we check the customization story before we do such? https://mui.com/x/react-date-pickers/custom-field/#using-material-textfield feels like how most people should customize the date picker. I mean, I would expect most people to come here and not use Material UI input but a custom one they have created. When we add `enableAccessibleFieldDOMStructure` it crashes:

https://github.com/user-attachments/assets/81bba22c-c96b-44c9-a856-ea6c1d63d56e
